### PR TITLE
emit S_UDT symbols to keep all UDT alive after the linker.

### PR DESCRIPTION
**Pull request by sandreenko**
We need s_udt to guarantee that all UDT will be alive after the linker.

@nattress PTAL.

### CODE REVIEW

File: lib/ObjWriter/objwriter.cpp [hunk 1]
```diff
@@ -702,9 +702,33 @@ void ObjectWriter::EmitDebugLoc(int NativeOffset, int FileId, int LineNumber,
   }
 }
 
+void ObjectWriter::EmitCVUserDefinedTypesSymbols() {
```
Comment (nattress) [inline lib/ObjWriter/objwriter.cpp]: Out of interest, what do we do if there are no UDTs? Should we skip this altogether and save the bytes?

**Comment by nattress** [comment]
What is the practical effect of this? What do we currently lose by not having this fix? It would be nice to have that rationale called out in the check-in comments so future spelunkers of the code know what the intent behind the fix is.

**Comment by sandreenko** [comment]
@nattress it fixes dotnet/corert#3538.

File: lib/ObjWriter/objwriter.cpp (hunk 1 above)
Comment (sandreenko) [inline lib/ObjWriter/objwriter.cpp]: Fixed

**Pull request merged**